### DISCUSSION TITLE
Add spectral dispersion to the ray tracing algorithm plus wavelength dependent optical properties for volumes and surfaces.

### DISCUSSION THREAD

**Discussion by jantma2** [mitsuba-renderer/mitsuba3#946] [Feature requests]
Is there any chance to add dispersion effects to the engine?

**Comment by njroussel** [comment]
Hi @jantma2 

This is technically feasible, although we have no current interest in implementing this ourselves.
Handling dispersion can come at a significant cost, there are cases where we'd most likely want to avoid that cost. I'd welcome a PR on this feature if there were some sort of a toggle to turn it on/off.
